packages/functions/src/context.ts

```diff
@@ -47,11 +48,16 @@ export class ContextProvider {
   private auth: FirebaseAuthInternal | null = null;
   private messaging: MessagingInternal | null = null;
   private appCheck: FirebaseAppCheckInternal | null = null;
+  private serverAppAppCheckToken: string | null = null;
   constructor(
+    readonly app: FirebaseApp,
     authProvider: Provider<FirebaseAuthInternalName>,
     messagingProvider: Provider<MessagingInternalComponentName>,
     appCheckProvider: Provider<AppCheckInternalComponentName>
   ) {
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
```

Comment:
Can't attach comment to line 85 but can we put a `?` there so it's `appCheckProvider?.get()` like RTDB and DataConnect etc seem to have done and I just noticed in this PR, for max efficiency in not bothering to even call `get()` in the case of no provider.

Reply:
Done.